docs/plugins.md

```diff
@@ -89,6 +89,13 @@ class GzipPlugin(MetadataPlugin):
 The same method can be used to, for example, automatically download and extract
 files from s3 automatically.
 
+One caveat worth calling out:
+Filter plugins are first executed in the context of daemon (when matching files),
+and later in the context of the web application (when downloading files from the UI).
+It's important that the set of filter plugins is the same in both cases.
+For example this means that all daemons must have the same plugins active.
+In most deployments this difference is not significant.
```

Comment:
What's the exact meaning of `In most deployments this difference is not significant.` here?
If it's similar to "it's very unlikely that you'll have to worry about" then maybe use `It's unlikely to happen in most deployments but be careful.`?

Reply:
I was wondering how to write this myself.

Basically, I don't think there's a single deployment at earth right now where this matters. To run into this you would have to:
 - have two or more different ursadb backends for a single mquery instance (this is already very unlikely. I've used it and it was proposed for CERT, but not sure anybody does it)
 - the backends would have to use a different sets of plugins. For example, one ursadb backend uses gzip compresion and one doesn't.

Spinning up a setup like this requires a good understanding of all the components and how they interact. It doesn't just happen by accident, and someone who did this will have an easy time debugging issues arising from divergent plugins.

So basically, this is just an acknowledgement that "we know that using multiple backends with different plugins is not well supported". 

With that in mind - do you think I should reword the warning?

Reply:
How about a simple `If your deployment doesn't have multiple ursadb backends (which is most likely the case) you don't have to worry about it`?

Reply:
> - have two or more different ursadb backends for a single mquery instance (this is already very unlikely. I've used it and it was proposed for CERT, but not sure anybody does it).  
> - the backends would have to use a different sets of plugins. For example, one ursadb backend uses gzip compresion and one doesn't.


Exactly our case :)

Reply:
Haha, I should never say "probably nobody does X" when talking about a software.

The problem is that plugins are a bit "dumb". For example imagine that you have one backend with gzip plugin (all files are ungzipped) and one backend without. Mquery then issues a search request, and gets a list of matches (it's just a list of paths/file IDs):

```
/backendA/aaaaa
/backendB/bbbbb (gzipped)
/backendA/cccccc
/backendB/bbbbb (gzipped)
```

Mquery must somehow know what to do with gzipped files, and which files are gzipped. There are at least two solutions:
 - make the plugin writer handle this. For example write a "smart gzip" that passthroughs files with paths/IDs `/backendA/`, and decompresses files with `/backendB/`.
 - handle this in mquery. Keep track of which plugins were used by the backend during a job, and decide in mquery which plugins to run on the basis. This will be a bit more flexible, but requires some more backend changes.

Reply:
What do you think about this version?